test/rekt/resources/account_role/role.yaml

```diff
@@ -18,19 +18,44 @@ metadata:
   name: {{ .name }}
   namespace: {{ .namespace }}
 
-{{ if .matchLabel }}
+{{- /* only print the ClusterRole if there are any rules or a matchLabel */ -}}
+{{ if or .matchLabel .rules }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: {{ .role }}
+
+{{ if .matchLabel }}
 aggregationRule:
   clusterRoleSelectors:
     - matchLabels:
         {{ .matchLabel }}: "true"
+{{ end }}
+
+{{ if .rules }}
+rules:
+  {{ range $_, $rule := .rules }}
```

Comment:
I wanted to extend and reuse the existing `account_role` here. 
In the future if we need to add more functionality here, I would create a separate template; but it is manageable for now.